recipes-kernel/linux/linux-imx.inc

```diff
@@ -4,19 +4,27 @@
 LICENSE = "GPLv2"
 LIC_FILES_CHKSUM = "file://COPYING;md5=d7810fab7487fb0aad327b76f1be7cd7"
 
-inherit kernel fsl-kernel-localversion fsl-vivante-kernel-driver-handler
+inherit kernel-yocto kernel fsl-kernel-localversion fsl-vivante-kernel-driver-handler
 
 # Put a local version until we have a true SRCREV to point to
 LOCALVERSION ?= ""
 SCMVERSION ?= "y"
 SRCBRANCH ?= ""
 
+# Set the PV to the correct kernel version to satisfy the kernel version sanity check
+PV = "${LINUX_VERSION}+git${SRCPV}"
+
 SRC_URI = "git://source.codeaurora.org/external/imx/linux-imx;protocol=https;branch=${SRCBRANCH} \
            file://defconfig \
 "
 
 S = "${WORKDIR}/git"
 
+# Tell to kernel class that we would like to use our defconfig to configure the kernel.
+# Otherwise, the --allnoconfig would be used per default which leads to mis-configured
+# kernel
+KCONFIG_MODE="--alldefconfig"
```

Comment:
Why this is needed?

Reply:
This is actually explained in the comment: `KCONFIG_MODE` (kind of) defaults to `--allnoconfig` in `kernel-yocto.bbclass` in function `do_kernel_configme()`:

```python
do_kernel_configme() {
        # translate the kconfig_mode into something that merge_config.sh
        # understands
        case ${KCONFIG_MODE} in
                *allnoconfig)
                        config_flags="-n"
                        ;;
                *alldefconfig)
                        config_flags=""
                        ;;
            *)
                if [ -f ${WORKDIR}/defconfig ]; then
                        config_flags="-n"
                fi
            ;;
        esac
< snip >
```

This is also explained in [meta-skeleton/recipes-kernel/linux/linux-yocto-custom.bb](https://git.yoctoproject.org/cgit/cgit.cgi/poky/tree/meta-skeleton/recipes-kernel/linux/linux-yocto-custom.bb#n34)

Reply:
TIL, thanks for that. 

I belive the comment should be extended to cover the below aspect: 

> If the defconfig file was created with make_savedefconfig, 
> not all options are specified, and should be restored with their
> defaults, not set to 'n'. To properly expand a defconfig like
> this, specify: KCONFIG_MODE="--alldefconfig" in the kernel
> recipe.

Reply:
> 
> 
> TIL, thanks for that.

👍 

> 
> I belive the comment should be extended to cover the below aspect:
> 
> > If the defconfig file was created with make_savedefconfig,
> > not all options are specified, and should be restored with their
> > defaults, not set to 'n'. To properly expand a defconfig like
> > this, specify: KCONFIG_MODE="--alldefconfig" in the kernel
> > recipe.

Yeap, this makes total sense! Would extend the comment with this description, since it is not really obvious.

Reply:
I've added a description block from `meta-skeleton/recipes-kernel/linux/linux-yocto-custom.bb`